plugins/network-elements/bigswitch/src/com/cloud/network/bigswitch/BigSwitchBcfApi.java

```diff
@@ -296,7 +296,7 @@ private void setHttpHeader(final HttpMethodBase m) {
         m.setRequestHeader(CONTENT_TYPE, CONTENT_JSON);
         m.setRequestHeader(ACCEPT, CONTENT_JSON);
         m.setRequestHeader(HTTP_HEADER_INSTANCE_ID, CLOUDSTACK_INSTANCE_ID + "-" + zoneId);
-        if (hash != "" ) {
+        if (! "".equals(hash)) {
```

Comment:
I was guessing that the code intents to set the value only when it is not null and not empty. with this code, when the hash value is null, it will still set the request header HASH_MATCH.

isNotEmpty impl, you should send PR to commons lib :) we use at multiple places